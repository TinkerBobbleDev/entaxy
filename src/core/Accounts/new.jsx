import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import AccountForm from './form'
import { createAccount } from '../../store/accounts/actions'

const mapDispatchToProps = {
  handleSave: account => createAccount(account)
}

const NewAccount = ({ history, handleSave }) => {
  const onSave = (account) => {
    handleSave(account)
    history.push(`/accounts/${account.id}/transactions`)
  }

  const onCancel = () => {
    history.push('/dashboard')
  }

  return (
    <AccountForm handleSave={onSave} handleCancel={onCancel} />
  )
}

NewAccount.propTypes = {
  history: PropTypes.object.isRequired,
  handleSave: PropTypes.func.isRequired
}

export default connect(null, mapDispatchToProps)(NewAccount)
