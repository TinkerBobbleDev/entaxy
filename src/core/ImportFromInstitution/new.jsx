import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import ImportFromInstitutionForm from './form'
import { createAccountGroup } from '../../store/accounts/actions'

const mapDispatchToProps = {
  createAccountGroup: (institution, accountGroup, accounts) =>
    createAccountGroup(institution, accountGroup, accounts)
}

export class NewImportFromInstitutionComponent extends React.Component {
  handleSave = (accountGroup, accounts) => {
    const { institution } = this.props.match.params
    this.props.createAccountGroup(institution, accountGroup, accounts)
    this.props.history.push('/dashboard')
  }

  handleCancel = () => {
    this.props.history.push('/dashboard')
  }

  render() {
    const { institution } = this.props.match.params
    return (
      <ImportFromInstitutionForm
        handleSave={this.handleSave}
        handleCancel={this.handleCancel}
        institution={institution}
      />
    )
  }
}

NewImportFromInstitutionComponent.propTypes = {
  history: PropTypes.object.isRequired,
  createAccountGroup: PropTypes.func.isRequired,
  match: PropTypes.object.isRequired
}

export default connect(null, mapDispatchToProps)(NewImportFromInstitutionComponent)
