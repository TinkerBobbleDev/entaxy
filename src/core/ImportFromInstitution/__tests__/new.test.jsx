import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { NewImportFromInstitutionComponent } from '../new'

jest.mock('../form', () => 'AccountForm')

describe('New Import From Institution', () => {
  const mochCreateAccountGroup = jest.fn()

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('matches snapshot', () => {
    const component = renderer.create((
      <NewImportFromInstitutionComponent
        createAccountGroup={mochCreateAccountGroup}
        history={{}}
        match={{ params: { institution: 'TD' } }}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  describe('Component methods', () => {
    const mochHistoryPush = jest.fn()
    const wrapper = shallow((
      <NewImportFromInstitutionComponent
        createAccountGroup={mochCreateAccountGroup}
        history={{ push: mochHistoryPush }}
        match={{ params: { institution: 'TD' } }}
      />
    ))
    const instance = wrapper.instance()

    it('saves the form', async () => {
      const accountGroup = { apiKey: 'abc', apiSecret: 'def' }
      const accounts = [{ name: 'account 1' }, { name: 'account 2' }]

      instance.handleSave(accountGroup, accounts)
      expect(mochCreateAccountGroup).toHaveBeenCalled()
      expect(mochHistoryPush).toHaveBeenCalledWith('/dashboard')
    })

    it('cancels the account creation', () => {
      instance.handleCancel()
      expect(mochHistoryPush).toHaveBeenCalledWith('/dashboard')
    })
  })
})
