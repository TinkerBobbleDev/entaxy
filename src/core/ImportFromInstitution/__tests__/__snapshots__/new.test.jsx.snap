// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Import From Institution matches snapshot 1`] = `
<AccountForm
  handleCancel={[Function]}
  handleSave={[Function]}
  institution="TD"
/>
`;
